common/src/java/org/apache/hadoop/hive/conf/HiveConf.java

```diff
@@ -101,6 +102,65 @@
   private volatile boolean isSparkConfigUpdated = false;
   private static final int LOG_PREFIX_LENGTH = 64;
 
+  interface HiveConfEnum<T extends Enum<T>> {
+    public static <T extends Enum<T>> T from(Class<T> cls, String value, T invalidEnum) {
+      try {
+        return T.valueOf(cls, value.toUpperCase());
+      } catch (Exception e) {
+        return invalidEnum;
+      }
+    }
+  }
```

Comment:
Should I even bother with this interface? The intent was originally to have static definitions of getValidSet and getInvalid as part of the interface in addition to from. Except Java interfaces don't allow implementations of an interface to override static methods. The from method here is also not super useful in that I think I still have to fully qualify the call (I.E. HiveConfEnum.from()). It would be nice if we could define required static methods on a enum interface for future uses of enums in HiveConf, but I don't think it is cleanly possible.

Reply:
I am not sure we need the interface indeed, probably the utility method would suffice?

Reply:
Yeah, I'll remove it. I don't even remember why it was a question for me.